src/client/dfuse/ops/open.c

```diff
@@ -100,7 +100,7 @@ dfuse_cb_open(fuse_req_t req, fuse_ino_t ino, struct fuse_file_info *fi)
 	/* Enable this for files up to the max read size. */
 	if (prefetch && oh->doh_parent_dir &&
 	    atomic_load_relaxed(&oh->doh_parent_dir->ie_linear_read) && ie->ie_stat.st_size > 0 &&
-	    ie->ie_stat.st_size <= DFUSE_MAX_READ) {
+	    ie->ie_stat.st_size <= DFUSE_MAX_PRE_READ) {
```

Comment:
Maybe I'm confused about what st_size is here.  Is it the file size?  Shouldn't readahead only happen if the size is >= DFUSE_MAX_PRE_READ ?  Otherwise, wouldn't it cause dfs to do a short read?

Reply:
ah, nevermind, this is just for small files

Reply:
Documentation for pre-read is at https://github.com/daos-stack/daos/blob/master/src/client/dfuse/dfuse.h#L97-L129

I have thought about extending this for larger files, as long as the file is read sequentially then the whole file could be treaded in a similar manner with reads being served from cache and and each read triggering the next one.  There's not a ticket for that yet though.

Reply:
This is separate from this ticket, but yea having a read-ahead on large files so future reads get served from cache is probably worth exploring.

Reply:
I agree, this is something we've discussed but really is a different feature from this.  I'll file a ticket.